applications/smart-distancing/libs/core.py

```diff
@@ -95,9 +95,57 @@ def process_video(self, video_uri):
     def process_image(self, image_path):
         # Process and pass the image to ui modules
         cv_image = cv.imread(image_path)
-        cv_image, objects, distancings = self.__process(cv_image)
+        cv_image, objects, distancings, _ = self.__process(cv_image)
         self.ui.update(cv_image, objects, distancings)
 
+    def process_image_export_results(self, image_path):
+        # Process and pass the image to ui module plus export results for evaluating
+        w = self.gt_image_size[0]
+        h = self.gt_image_size[1]
+        results_directory = self.config.get_section_dict('Evaluation')['ResultDir']
+        if not os.path.exists(results_directory):
+            os.mkdir(results_directory)
+        path = os.path.join(results_directory,
+                            self.device + '-' + self.config.get_section_dict('Detector')['Name'])
+        if not os.path.exists(path):
+            os.mkdir(path)
+
+        for filename in os.listdir(image_path):
+            results = ''
+            dt_fromat = "{0} {1} {2} {3} {4} {5} \n"
+            if not (filename.endswith('.jpg') or filename.endswith('jpeg')): continue
+            img_path = os.path.join(image_path, filename)
+            cv_image = cv.imread(img_path)
+            cv_image, objects, distancings, original_boxes = self.__process(cv_image)
+            # print(original_boxes)
+            image_name = filename.split('.')[0]
+            for obj in original_boxes:
+                # category_id = obj['id'].split('-')[0]
+                class_name = 'pedestrian'  # TODO: category json
+                bbox = obj['bbox']
+                # Do modifications
+                bbox[0] = np.maximum(0, bbox[0])
+                bbox[1] = np.maximum(0, bbox[1])
+                bbox[2] = np.minimum(w, bbox[2])
+                bbox[3] = np.minimum(h, bbox[3])
+                x0 = bbox[0]
+                y0 = bbox[1]
+                x1 = bbox[2]
+                y1 = bbox[3]
+                x_abs = x0 * w
+                y_abs = y0 * h
+                width = (x1 * w) - x_abs
+                height = (y1 * h) - y_abs
```

Comment:
I think you don't need to compute these values since they exist in one of the `objects_list` values.

Reply:
To the best of my knowledge some modifications are applied to the objects_list bbox such as removing small/large bbox and ... which is impacted on evaluation metrics!